client/rpc/src/offchain/mod.rs

```diff
@@ -0,0 +1,67 @@
+// Copyright 2017-2020 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Substrate offchain API.
+
+#[cfg(test)]
+mod tests;
+
+/// Re-export the API for backward compatibility.
+pub use sc_rpc_api::offchain::*;
+use self::error::{Error, Result};
+use sp_core::{
+	Bytes,
+	offchain::{OffchainStorage, StorageKind},
+};
+use parking_lot::RwLock;
+use std::sync::Arc;
+
+/// Offchain API
+#[derive(Debug)]
+pub struct Offchain<T: OffchainStorage> {
+	/// Offchain storage
+	storage: Arc<RwLock<T>>,
+}
+
+impl<T: OffchainStorage> Offchain<T> {
+	/// Create new instance of Offchain API.
+	pub fn new(storage: T) -> Self {
+		Offchain {
+			storage: Arc::new(RwLock::new(storage)),
+		}
+	}
+}
+
+impl<T: OffchainStorage + 'static> OffchainApi for Offchain<T> {
+	/// Set offchain local storage under given key and prefix.
+	fn set_local_storage(&self, kind: StorageKind, key: Bytes, value: Bytes) -> Result<()> {
+		let prefix = match kind {
+			StorageKind::PERSISTENT => sp_offchain::STORAGE_PREFIX,
+			StorageKind::LOCAL => return Err(Error::UnavailableStorageKind),
```

Comment:
Should this be supported in the future? If not, I don't understand why we take `kind` here.

Reply:
Yes, it mirrors the API available for offchain workers. Implementation could be done on top of #3263

Reply:
What is the difference between local and persistent?

Reply:
Fork awarenes. `LOCAL` entries should be removed in case there is an reorg, while `PERSISTENT` should stay even if reorg happened (i.e. we see results produced by offchain workers that run on a different fork). The second one is useful if you want to make sure to avoid duplication of external "side effects".